api/api/views/media_views.py

```diff
@@ -265,16 +277,30 @@ def report(self, request, identifier):
 
         return Response(data=serializer.data, status=status.HTTP_201_CREATED)
 
-    def thumbnail(self, request, media_obj, image_url):
+    async def get_image_proxy_media_info(self) -> image_proxy.ImageProxyMediaInfo:
+        raise NotImplementedError(
+            "Subclasses must implement `get_image_proxy_media_info`"
+        )
+
+    @action(
+        detail=True,
+        url_path="thumb",
+        url_name="thumb",
+        serializer_class=media_serializers.MediaThumbnailRequestSerializer,
+        throttle_classes=[AnonThumbnailRateThrottle, OAuth2IdThumbnailRateThrottle],
```

Comment:
todo: confirm that throttles are inherited on this view. If not, refactor to put explicit throttles back.